### PR TITLE
Abstract maven calls

### PR DESCRIPTION
**Pull request by Meggolbek**
Invariants are still correct. 😄

### CODE REVIEW

File: getty/center.py [hunk 1]
```diff
@@ -278,45 +278,50 @@ def get_expansion_set(go):
...
-    os.sys_call("mvn clean")
+    # os.sys_call("git checkout " + this_hash)
+    # os.sys_call("mvn clean")
 
-    bin_path = mvn.path_from_mvn_call("outputDirectory")
-    test_bin_path = mvn.path_from_mvn_call("testOutputDirectory")
-    cp = mvn.full_classpath(junit_path, sys_classpath, bin_path, test_bin_path)
+    bin_path = mavenCalls.get_bin_path(this_hash)
+    test_bin_path = mavenCalls.get_test_bin_path(this_hash)
+    cp = mavenCalls.get_full_class_path(this_hash, junit_path, sys_classpath, bin_path, test_bin_path)
```
Comment (sandervalstar) [inline getty/center.py]: can you call it "mvn_adapter" instead of mavenCalls?

File: getty/center.py [hunk 2]
```diff
@@ -278,45 +278,50 @@ def get_expansion_set(go):
...
     if SHOW_DEBUG_INFO:
         print "\n===full classpath===\n" + cp + "\n"
-    
+
     print "\ncopying all code to specific directory ...\n"
-    all_code_dirs = [mvn.path_from_mvn_call("sourceDirectory"),
-                     # mvn.path_from_mvn_call("scriptSourceDirectory"),
-                     mvn.path_from_mvn_call("testSourceDirectory")]
+    start = time.time()
```
Comment (sandervalstar) [inline getty/center.py]: timer can be removed?

File: getty/tools/mavenCalls.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import mvn, os
+
+def get_bin_path(hash):
+    os.sys_call("git checkout " + hash)
```
Comment (sandervalstar) [inline getty/tools/mavenCalls.py]: can you add a "git_adapter" or perhaps just "git" with a checkout(hash) method?

File: getty/tools/mavenCalls.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    maven_clean()
+    return mvn.full_classpath(junit_path, sys_classpath, bin_output, test_output)
+
+def compile_tests(hash):
+    os.sys_call("git checkout " + hash)
+    maven_clean()
+    os.sys_call("mvn test-compile")
+
+def get_Junit_torun(cust_mvn_repo, hash):
```
Comment (sandervalstar) [inline getty/tools/mavenCalls.py]: Junit->junit

File: getty/center.py (hunk 1 above)
Comment (Meggolbek) [reply getty/center.py]: done!

File: getty/center.py (hunk 2 above)
Comment (Meggolbek) [reply getty/center.py]: yep! Done!

File: getty/tools/mavenCalls.py (hunk 4 above)
Comment (Meggolbek) [reply getty/tools/mavenCalls.py]: fixed!

File: getty/tools/mavenCalls.py (hunk 3 above)
Comment (Meggolbek) [reply getty/tools/mavenCalls.py]: done!

**Pull request closed without merging**